lib/date.js

```diff
@@ -1,215 +1,228 @@
 };
 
 CronDate.prototype.toJSON = function() {
   return this._date.toJSON();
 };
 
 CronDate.prototype.setDate = function(d) {
-  return this._date.date(d);
+  this._date = this._date.set({ day: d });
 };
 
 CronDate.prototype.setFullYear = function(y) {
-  return this._date.year(y);
+  this._date = this._date.set({ year: y });
 };
 
 CronDate.prototype.setDay = function(d) {
-  return this._date.day(d);
+  this._date = this._date.set({ weekday: d });
 };
 
 CronDate.prototype.setMonth = function(m) {
-  return this._date.month(m);
+  this._date = this._date.set({ month: m + 1 });
 };
 
 CronDate.prototype.setHours = function(h) {
-  return this._date.hour(h);
+  this._date = this._date.set({ hour: h });
 };
 
 CronDate.prototype.setMinutes = function(m) {
-  return this._date.minute(m);
+  this._date = this._date.set({ minute: m });
 };
 
 CronDate.prototype.setSeconds = function(s) {
-  return this._date.second(s);
+  this._date = this._date.set({ second: s });
 };
 
 CronDate.prototype.setMilliseconds = function(s) {
-  return this._date.millisecond(s);
-};
-
-CronDate.prototype.getTime = function() {
-  return this._date.valueOf();
+  this._date = this._date.set({ millisecond: s });
 };
 
 CronDate.prototype._getUTC = function() {
-  return moment.utc(this._date);
+  return this._date.toUTC();
 };
 
 CronDate.prototype.toString = function() {
   return this._date.toString();
 };
 
 CronDate.prototype.toDate = function() {
-  return this._date.toDate();
+  return this._date.toJSDate();
 };
 
 CronDate.prototype.isLastDayOfMonth = function() {
-  var newDate = this._date.clone();
   //next day
-  newDate.add(1, 'day').startOf('day');
-  return this._date.month() !== newDate.month();
+  var newDate = this._date.plus({ days: 1 }).startOf('day')
+  return this._date.month !== newDate.month;
 };
 
 function CronDate (timestamp, tz) {
-  if (timestamp instanceof CronDate) {
-    timestamp = timestamp._date;
+  var dateOpts = { zone: tz };
+  if (!timestamp) {
+    this._date = luxon.DateTime.local()
+  } else if (timestamp instanceof CronDate) {
+    this._date = timestamp._date;
+  } else if (timestamp instanceof Date) {
+    this._date = luxon.DateTime.fromJSDate(timestamp, dateOpts);
+  } else if (typeof timestamp === "number") {
+    this._date = luxon.DateTime.fromMillis(timestamp, dateOpts);
+  } else if (typeof timestamp === "string") {
+    this._date = luxon.DateTime.fromISO(timestamp, dateOpts);
+    this._date.isValid || (this._date = luxon.DateTime.fromRFC2822(timestamp, dateOpts));
+    this._date.isValid || (this._date = luxon.DateTime.fromFormat(timestamp, "yyyy-MM-dd HH:mm:ss", dateOpts));
+    this._date.isValid || (this._date = luxon.DateTime.fromFormat(timestamp, "EEE, d MMM yyyy HH:mm:ss", dateOpts));
+    this._date.isValid || (this._date = luxon.DateTime.fromFormat(timestamp, "yyyy-MM-d", dateOpts));
+    this._date.isValid || (this._date = luxon.DateTime.fromFormat(timestamp, "yyyy-M-dd", dateOpts));
```

Comment:
```suggestion
  if (!timestamp) {
    this._date = luxon.DateTime.local()
  } else if (timestamp instanceof CronDate) {
    this._date = timestamp._date;
  } else if (timestamp instanceof Date) {
    this._date = luxon.DateTime.fromJSDate(timestamp, dateOpts);
  } else if (typeof timestamp === 'number') {
    this._date = luxon.DateTime.fromMillis(timestamp, dateOpts);
  } else if (typeof timestamp === 'string') {
    this._date = luxon.DateTime.fromISO(timestamp, dateOpts);
    this._date.isValid || (this._date = luxon.DateTime.fromRFC2822(timestamp, dateOpts));
    this._date.isValid || (this._date = luxon.DateTime.fromFormat(timestamp, 'yyyy-MM-dd HH:mm:ss', dateOpts));
    this._date.isValid || (this._date = luxon.DateTime.fromFormat(timestamp, 'EEE, d MMM yyyy HH:mm:ss', dateOpts));
    this._date.isValid || (this._date = luxon.DateTime.fromFormat(timestamp, 'yyyy-MM-d', dateOpts));
    this._date.isValid || (this._date = luxon.DateTime.fromFormat(timestamp, 'yyyy-M-dd', dateOpts));
```

Single-quotes are mostly preferred.

It would be also nice if we could have some tests this updated input date handling.

Reply:
done